internal/output/output.go

```diff
@@ -198,6 +199,9 @@ func (o Output) Violations() []output.Result {
 	violations = o.AttestationSyntaxCheck.addToViolations(violations)
 	violations = o.addCheckResultsToViolations(violations)
 
+	sort.Slice(violations, func(i, j int) bool {
+		return violations[i].Message < violations[j].Message
```

Comment:
Why not sort by the "code" value here also? Or maybe by the code first and message second?

Reply:
@simonbaird `code` seems to be less descriptive for violations than other types. For example, when testing `ec-cli` against `quay.io/cuipinghuo/hermetic_build:0.3`, I get:

```
- msg: Pipeline task 'appstudio-configure-build' does not contain a bundle reference
  metadata:
    code: attestation_task_bundle.disallowed_task_reference
    effective_on: "2022-01-01T00:00:00Z"
- msg: Pipeline task 'appstudio-init' does not contain a bundle reference
  metadata:
    code: attestation_task_bundle.disallowed_task_reference
    effective_on: "2022-01-01T00:00:00Z"
- msg: Pipeline task 'build-container' does not contain a bundle reference
  metadata:
    code: attestation_task_bundle.disallowed_task_reference
    effective_on: "2022-01-01T00:00:00Z"
- msg: Pipeline task 'clair-scan' does not contain a bundle reference
  metadata:
    code: attestation_task_bundle.disallowed_task_reference
    effective_on: "2022-01-01T00:00:00Z"
...
```

Note that all `code` values are the same. In practice, sorting on that value _still_ isn't deterministic - the task names will be mixed up from run to run.

Sorting against `msg`, on the other hand, captures these differences and tends to be more deterministic.

I have tried sorting on both criteria using the `sort.Slice` function sequentially, but the end result seemed to be that the later sort just overwrote the first one, making it redundant. Perhaps there's a much smarter sort method in Go I'm unaware of - if anyone happens to know one, please chime in :)

Compare and contrast that to the success results:

```
- msg: Pass
  metadata:
    code: attestation_task_bundle.empty_task_bundle_reference
- msg: Pass
  metadata:
    code: attestation_task_bundle.missing_required_data
- msg: Pass
  metadata:
    code: attestation_task_bundle.out_of_date_task_bundle
```

In this case, `msg` doesn't tell us anything, but `metadata.code` does - so it makes sense to sort on `metadata.code` for success.

I don't have enough example output for "warnings" to make a call one way or the other, though.

Reply:
There's some value in sorting by code first and msg second. I think it would be a little weird if violations with the same "code" value were not together in the sorted output.

Sorting by two fields should be doable. You have to put the logic in the sort function. Let me try to demonstrate.